### PR TITLE
IPMPROG-4490: improve resilience

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**

### CODE REVIEW

**Review by ericclappier-eaton — APPROVED** [review]

File: asset/src/asset-db.cpp [hunk 1]
```diff
@@ -471,6 +474,16 @@ Expected<Attributes> selectExtAttributes(uint32_t elementId)
             attrs.emplace(row.get("keytag"), val);
         }
 
+        // IPMPROG-4490: [resilience]
+        // If no endpoint.1 port defined, set default related to the protocol
+        if ((attrs.count("endpoint.1.port") == 0) && (attrs.count("endpoint.1.protocol") != 0)) {
```
Comment (ericclappier-eaton) [inline asset/src/asset-db.cpp:479]: Same for endpoint.2.port ?

File: asset/src/db/select.cpp [hunk 2]
```diff
@@ -589,13 +593,22 @@ Expected<Attributes> extAttributes(Connection& conn, uint32_t elementId)
...
-
             row.get("value", val.value);
             row.get("read_only", val.readOnly);
 
             attrs.emplace(row.get("keytag"), val);
         }
 
+        // IPMPROG-4490: [resilience]
+        // If no endpoint.1 port defined, set default related to the protocol
+        if ((attrs.count("endpoint.1.port") == 0) && (attrs.count("endpoint.1.protocol") != 0)) {
```
Comment (ericclappier-eaton) [inline asset/src/db/select.cpp:604]: Same for endpoint.2.port ?

File: asset/src/asset-db.cpp (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply asset/src/asset-db.cpp:479]: no... its optional related to mass management

File: asset/src/db/select.cpp (hunk 2 above)
Comment (FrancoisRegisDegott-eaton) [reply asset/src/db/select.cpp:604]: no... its optional related to mass management

**Review by ericclappier-eaton — APPROVED** [review]

**Pull request merged**